tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/deploy-stress-test-resources.ps1

```diff
@@ -15,6 +15,10 @@ mkdir /azure
 Copy-Item "/scripts/stress-test/test-resources-post.ps1" -Destination "/azure/"
 Copy-Item "/mnt/testresources/*" -Destination "/azure/"
 
+while (kubectl get pods -n $env:namespace $env:BASE_NAME -o jsonpath='{.items[?(@.metadata.annotations.batch\.kubernetes\.io/job-completion-index=="0")].metadata.name}' -nq "Complete") {
```

Comment:
Does this work? Usually the default pod service identity doesn't have GET access to the k8s api (unless I'm thinking of cross-namespace requests). Also how does the pod with index 0 complete if it reads its own status? Do we need an extra skip condition for index 0?

Reply:
That said, I like this approach much more than polling for ARM deployment completion, pretty clever.

Reply:
Also `-ne` not `-nq`